### PR TITLE
test: do not trigger node bug with fd mapping

### PR DESCRIPTION
**Pull request by sam-github**
See https://github.com/iojs/io.js/issues/862

Note this is not enough to fix the tests, more subtle problems are causing the test failures for the express records.

/to @rmg this should effect tap, AFAICT.

/cc @bajtos

### CODE REVIEW

**Comment by sam-github** [comment]
I take it back, the problems weren't subtle, its just they were obscured by the node bug. Anyhow, I hope to see a green up above RSN.

File: bin/sl-run.js [hunk 1]
```diff
@@ -11,15 +11,42 @@ process.on('disconnect', function() {
...
-  if(config.isMaster) {
-    log.error('supervisor running without StrongOps (unprofiled)');
-  }
-} else {
-  require('strong-agent').profile(undefined, undefined, {
-    quiet: config.isWorker, // Quiet in worker, to avoid repeated log messages
-    logger: config.logger,
-  });
+var agent = require('../lib/agent');
+var options = {
```
Comment (rmg) [inline bin/sl-run.js]: It's really really minor, but could this be `agentOptions` to make the association more clear?

File: bin/sl-run.js [hunk 2]
```diff
@@ -11,15 +11,42 @@ process.on('disconnect', function() {
...
+    // Otherwise, just start. This is a no-op if it is already started.
+    agent().start();
+    break;
+
+  case true: // Profiling or metrics explicitly enabled.
+    agent().configure(options);
+    // Only try to start StrongOps if they have registered, to avoid legacy
+    // warning messages. If an app is missing a name, profile may still fail
+    // to start, so drop-through to start(). We must re-supply options.
+    if (agent().config.key)
```
Comment (rmg) [inline bin/sl-run.js]: Missing `{` `}`

**Comment by rmg** [comment]
Before this PR I hadn't considered whether JS's `switch` did coercion or not.

TIL: JavaScript uses `===` for `switch` statements and therefore does not coerce. Handy!

Interestingly (to me, at least), Ruby _also_ uses a `===` operator for its switch statement equivalent (`case` expressions), but the operator has completely different semantics.

**Comment by rmg** [comment]
Oh ya.. LGTM, other than a couple very minor style nits.

**Comment by bajtos** [comment]
:+1:

Thanks for fixing this @sam-github

File: bin/sl-run.js (hunk 1 above)
Comment (sam-github) [inline bin/sl-run.js]: yes

File: bin/sl-run.js (hunk 2 above)
Comment (sam-github) [inline bin/sl-run.js]: Not required! We really need lint rules, I guess. Note the existing code base, some written by you, does not always use `{}` for single-statement if: https://github.com/strongloop/strong-supervisor/blob/868ba2ce8cdd2d407c8244a4d2455206953935da/lib/expander.js#L11-L18

**Comment by sam-github** [comment]
I was a bit worried about the switch coercion, but I tested, and the docs are explicit, I'm glad it works!

**Pull request merged**